samples/SortBenchmark/SortBenchmark.fs

```diff
@@ -390,98 +266,310 @@ type RemoteFunc( filePartNum:int, records:int64, _dim:int , partNumS1:int, partN
 
         (parti, genLength)
 
-    member val repartitionThread = ref 0
 
 
-    member x.Repartitionfake parti serial (buffer:byte[], size:int)=
-        
-        Logger.LogF( LogLevel.WildVerbose, (fun _ -> sprintf "!!!repartition data!"))
-        let r = seq {
-                        //let t1 = DateTime.UtcNow
-                        for i = 0 to partNumS1 do
-                            Logger.LogF( LogLevel.WildVerbose, (fun _ -> sprintf "repartition data!"))
-                            let t = Array.zeroCreate<byte> (size / 1024)
-                            Buffer.BlockCopy(buffer,0,t,0,t.Length)
-                            yield i,t
-                        //let t2 = DateTime.UtcNow
-                        //Logger.LogF( LogLevel.WildVerbose, (fun _ -> sprintf "repartition: %d records, takes %f s; # repartition threads: %d" (size / 100) ((t2-t1).TotalSeconds) !x.repartitionThread))
 
-                        }
-        r
 
-    member x.ReadFilesfake parti= 
+
+    member x.GetReadFileBuf() = 
+            let byt = ref Unchecked.defaultof<_>
+            let bnewbuf = ref false
+            if not (RemoteFunc.sharedMem.TryDequeue(byt)) then
+                if (Interlocked.Increment(RemoteFunc.sharedMemSize) < 300) then
```

Comment:
General comment: There's no need for the extra indirection with "ref". So long as you don't capture the variables (byt and bnewbuf) in closures, you can use just "let mutable" and "&byt" when passing to TryDeque. Probably doesn't make much difference in most code, but just fyi.